### DISCUSSION TITLE
Security Vulnerabilities

### DISCUSSION THREAD

**Discussion by rook1337** [open-webui/open-webui#4031] [General]
Dear team,
Please check full detailed report here
https://huntr.com/bounties/2ed02d49-aa91-4061-98b9-61ce096f0396
https://huntr.com/bounties/94a68137-bdea-4e6f-8c20-19f3f1e6f0b8
https://huntr.com/bounties/f42cf72a-8015-44a6-81a9-c6332ef05afc
https://huntr.com/bounties/e449567a-01eb-4129-85a8-cd89e6cfcaa8

Please check and validate these vulnerabilities

**Comment by justinh-rahb** [comment]
I highly encourage you to instead use GitHub security disclosures, as those above seem to require an account. It's not very polite.

**Comment by rook1337** [comment]
hello @justinh-rahb 
Open-webui have hosted their bug bounty program on huntr.com so that we security researchers can report vulnerabilities directly on huntr.com. Only maintainer's can access these reports on huntr.com because of security reasons

**Comment by justinh-rahb** [comment]
Open WebUI hasn't got a bug bounty program. This seems sus. Interesting sales funnel though

![Screenshot 2024-07-21 at 12 46 18 PM](https://github.com/user-attachments/assets/2085ba48-0683-426d-abdb-9b41a5bcd46a)

**Reply by pressdarling** [reply]
Not the first time, looks like: https://github.com/stitionai/devika/issues/431 - but despite the comment on the platform, there are some reports the user has raised that have led to fixes or at least further discussion e.g. https://github.com/mlflow/mlflow/issues/11308, https://github.com/zenml-io/zenml/issues/2488.

But without a PR, I think [this is the right approach](https://github.com/run-llama/llama_index/issues/10487#issuecomment-1943118026) to these requests:
 
> I have not had time to even think about this. I get to it at some point. We are well aware your team has reported both known issues and other benign issues.
> 
> Hot take, but it's an open source repo, highly recommend making a PR to solve issues as well :)

**Reply by ricky-davis** [reply]
This does seem really sketchy and extremely misleading. Best case, huntr.com is adding projects without permission, providing false hope to bug hunters, and holding bugs hostage from the maintainers unless they join their platform. rook1337 is clearly under the assumption open-webui willingly joined this unnecessary program.
Who is funding these bug bounties? Because the devs of these projects clearly aren't.

**Comment by justinh-rahb** [comment]
We appreciate your report, but we want to clarify our stance. We won't be registering on a platform that requires a login, especially one that feels coercive. @rook1337, if you're seeking our validation for these vulnerabilities in exchange for payment from this platform, I'm afraid that isn't going to happen. We encourage you to submit pull requests if you have potential solutions, but we strongly believe that we should not have to pay or sign up on someone else's platform for this information. It feels unjust to us.

**Comment by rook1337** [comment]
Hello @justinh-rahb and @ricky-davis 
I am reporting vulnerabilities to the many programs and they are fixing it without any issue using huntr.com platform. As you know our end goal is to protect and make this repo secure by reporting vulnerabilities. I am doing all this for my living and i cannot report vulnerabilities for free. 
It is maintainer choice to validate the vulnerability within 45 days as per platform rules because the reports will be automatically will be published after 45 days.

Hope you all understand

Thanks

**Reply by justinh-rahb** [reply]
Guess we'll see in 45 days, thanks boss.

**Reply by rook1337** [reply]
BTW, I have checked with one of the program maintainer and they told me huntr.com is not charging them or taking vulnerabilities at hostage. I dont know from where you got that information

**Reply by justinh-rahb** [reply]
The participation in this scheme feels coerced and contradicts the principles of open-source collaboration. Our project has never communicated with Huntr.com and did not request this type of exposure. We certainly do not appreciate the insinuation that we have implicitly consented to participate in a bug bounty program. Such assumptions are not just misleading—they're fundamentally dishonest.

Moreover, suggesting or implying that our project maintains a bug bounty program without our explicit agreement is unacceptable. From what little we've unearthed, this seems to be part of a broader and more concerning practice. There are more ethical ways to engage in security-related activities, and this approach does a disservice to the community and principles of transparency and mutual respect that underpin open-source development.

**Reply by rook1337** [reply]
No problem, i will still get the bounty after 45 days. For registering the CVE, i will send the reports after 45 days on your Security Page.

**Reply by pressdarling** [reply]
@rook1337 This seems more like blackmail than good-faith collaboration, and regardless of whether that is coming from your own values or those of the huntr.com platform, I would recommend revising your own personal approach to bug bounties to avoid that perception while engaging with open-source projects.

Assuming that you are doing this in good faith, if you're chasing bounties for a living, there are certainly more lucrative fish in the pond.

**Comment by rook1337** [comment]
@justinh-rahb,
I have reported the vulnerabilities in your Github Security page. But no one is responding on them from past 2 week. Please check
